From bf0262170b9b02149d5e8d3266bd1e86b03b5650 Mon Sep 17 00:00:00 2001
From: Barry Warsaw <barry@python.org>
Date: Thu, 16 Jun 2016 15:29:12 +0300
Subject: Warn and skip broken requirements.

Description: Warn and skip broken requirements.
Origin: https://github.com/pypa/pip/pull/3704
Bug: https://github.com/pypa/pip/issues/3681
Forwarded: not-needed

Patch-Name: skip-bad-requirements.patch
---
 pip/operations/freeze.py | 16 ++++++++++++----
 1 file changed, 12 insertions(+), 4 deletions(-)

diff --git a/pip/operations/freeze.py b/pip/operations/freeze.py
index 086922e..7493ced 100644
--- a/pip/operations/freeze.py
+++ b/pip/operations/freeze.py
@@ -8,6 +8,7 @@ from pip.req import InstallRequirement
 from pip.utils import get_installed_distributions
 from pip._vendor import pkg_resources
 from pip._vendor.packaging.utils import canonicalize_name
+from pip._vendor.pkg_resources import RequirementParseError
 
 
 logger = logging.getLogger(__name__)
@@ -42,10 +43,17 @@ def freeze(
     for dist in get_installed_distributions(local_only=local_only,
                                             skip=(),
                                             user_only=user_only):
-        req = pip.FrozenRequirement.from_dist(
-            dist,
-            dependency_links
-        )
+        try:
+            req = pip.FrozenRequirement.from_dist(
+                dist,
+                dependency_links
+            )
+        except RequirementParseError:
+            logger.warning(
+                "Could not parse requirement: %s",
+                dist.project_name
+            )
+            continue
         installations[req.name] = req
 
     if requirement:
